src/alda/lisp/model/pitch.clj

```diff
@@ -27,14 +27,18 @@
 (defn determine-midi-note
   "Determines the MIDI note number of a note, within the context of an
    instrument's octave and key signature."
-  [{:keys [letter accidentals] :as note} octave key-sig]
-  (reduce (fn [number accidental]
+  [{:keys [letter accidentals] :as note} octave key-sig transpose]
+
+  ; check if note value in proper range after transpose
+  {:post [<= 0 % 127]}
```

Comment:
I think we should take this post-condition out. It would be a good idea if we were restricting ourselves to MIDI (...which we currently are), but I would like to leave it open so that in the future, we can add other types of audio-generating instruments that are not limited to the note range corresponding to MIDI numbers 0-127.